### PR TITLE
Added reusable side menu

### PR DESCRIPTION
**Pull request by Ymkavi**
Screenshot 2022-08-10 190313
Screenshot 2022-08-10 190454

### CODE REVIEW

File: frontend/src/app/shared/components/side-menu/side-menu-header/side-menu-header.component.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { SideMenuItemComponent } from '../side-menu-item/side-menu-item.component';
+
+@Component({
+    selector: 'app-side-menu-header',
+    templateUrl: './side-menu-header.component.html',
+    styleUrls: ['./side-menu-header.component.sass'],
+    providers: [
+        {
+            provide: SideMenuItemComponent,
+            // eslint-disable-next-line no-use-before-define
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/side-menu/side-menu-header/side-menu-header.component.ts]: Do not use eslint-disable

File: frontend/src/app/shared/components/side-menu/side-menu-item/side-menu-item.component.sass [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        border-radius: 4px
+        &:hover
+            background-color: $athens-gray
+            cursor: pointer
+
+.selected
+    pointer-events: none
+    color: $royal-blue
+    font-weight: 700
+    background-color: #E3ECFF
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/side-menu/side-menu-item/side-menu-item.component.sass]: Use our color variables

File: frontend/src/app/shared/components/side-menu/side-menu-item/side-menu-item.component.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    selector: 'app-side-menu-item',
+    templateUrl: './side-menu-item.component.html',
+    styleUrls: ['./side-menu-item.component.sass'],
+})
+export class SideMenuItemComponent implements AfterContentInit {
+    public isSelected: boolean = false;
+
+    @ViewChild(TemplateRef) template: TemplateRef<any>;
+
+    // eslint-disable-next-line no-use-before-define
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/side-menu/side-menu-item/side-menu-item.component.ts]: Do not use eslint-disable

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — CHANGES_REQUESTED** [review]
Need more discussion about logic on a call

File: frontend/src/app/modules/side-menu/component/side-menu.component.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<mat-nav-list>
+    <ng-container *ngFor="let group of groups">
+        <div class="group-header" *ngIf="group.header" mat-subheader>{{ group.header }}</div>
+        <a
+            mat-list-item
+            *ngFor="let item of group.items"
+            [routerLink]="item.routerLink"
+            routerLinkActive="side-list-item-active"
+        >
```
Comment (tatianahutii) [inline frontend/src/app/modules/side-menu/component/side-menu.component.html]:
```suggestion
            routerLinkActive="side-list-item-active">
```

File: frontend/src/app/modules/side-menu/component/side-menu.component.sass [hunk 5]
```diff
@@ -0,0 +1,16 @@
+
```
Comment (tatianahutii) [inline frontend/src/app/modules/side-menu/component/side-menu.component.sass]: Remove empty line

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Comment by Ymkavi** [comment]
Screenshot 2022-08-11 152438

File: frontend/src/app/modules/side-menu/component/side-menu.component.sass [hunk 6]
```diff
@@ -1,16 +1,20 @@
-
 @import 'src/styles/color-variables.sass'
 
 .group-header
     color: $loblolly
 
 .mat-list-base .mat-list-item, .mat-list-base .mat-subheader
     height: 1em
+    width: 259px
```
Comment (tatianahutii) [inline frontend/src/app/modules/side-menu/component/side-menu.component.sass]: This is not a good idea to use constant px

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]
LGTM

**Pull request merged**